### PR TITLE
Rename "DSL" to "Streamlet API"

### PR DESCRIPTION
**Pull request by lucperkins**
The following test has been completed successfully locally:

```bash
$ bazel build --config=darwin heron/... heronpy/... examples/...
```

Waiting on CI results but should pass AFAIK.

### CODE REVIEW

File: heron/api/src/java/com/twitter/heron/streamlet/SerializableGenerator.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//  Copyright 2017 Twitter. All rights reserved.
```
Comment (srkukarni) [inline heron/api/src/java/com/twitter/heron/streamlet/SerializableGenerator.java]: Whats this new file?

File: heron/api/src/java/com/twitter/heron/streamlet/impl/sources/GeneratorSource.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+//  Copyright 2017 Twitter. All rights reserved.
```
Comment (srkukarni) [inline heron/api/src/java/com/twitter/heron/streamlet/impl/sources/GeneratorSource.java]: same as above comment

File: heron/api/src/java/com/twitter/heron/streamlet/SerializableGenerator.java (hunk 1 above)
Comment (lucperkins) [reply heron/api/src/java/com/twitter/heron/streamlet/SerializableGenerator.java]: Both of these files (along with two others) seem to be results of errant merges.

File: heron/api/tests/java/BUILD [hunk 3]
```diff
@@ -26,9 +26,8 @@ java_tests(
     "com.twitter.heron.api.metric.CountStatAndMetricTest",
     "com.twitter.heron.api.metric.LatencyStatAndMetricTest",
     "com.twitter.heron.api.bolt.BaseWindowedBoltTest",
-    "com.twitter.heron.dsl.impl.BaseStreamletTest",
-    "com.twitter.heron.dsl.impl.BaseKVStreamletTest",
-    "com.twitter.heron.dsl.impl.operators.JoinOperatorTest"
+    "com.twitter.heron.streamlet.impl.BaseStreamletTest",
+    "com.twitter.heron.streamlet.impl.BaseKVStreamletTest",
```
Comment (srkukarni) [inline heron/api/tests/java/BUILD]: Need to include JoinOperatortest?

File: heronpy/streamlet/BUILD [hunk 4]
```diff
@@ -3,7 +3,7 @@ package(default_visibility = ["//visibility:public"])
 load("/tools/rules/pex/pex_rules", "pex_library", "pex_binary")
 
 pex_library(
-    name = "heron-pythondsl-py",
+    name = "heron-python-streamlet-api-py",
```
Comment (srkukarni) [inline heronpy/streamlet/BUILD]: why streamletapi? why not just streamlet?

File: heron/api/tests/java/BUILD (hunk 3 above)
Comment (lucperkins) [reply heron/api/tests/java/BUILD]: Done

File: heronpy/streamlet/BUILD (hunk 4 above)
Comment (lucperkins) [reply heronpy/streamlet/BUILD]: Works for me either way. I'll shorten it to just streamlet.

**Pull request merged**